### ISSUE TITLE
N+1 with nested validation

### ISSUE THREAD

**Issue by eric-simonton-sama** [zdennis/activerecord-import#488]
I am importing records recursively, and a nested record has a validation that uses one of its other associations. Something like this:
```ruby
class Parent < ApplicationRecord
  has_many: children
end

class Child < ApplicationRecord
  belongs_to: parent
  belongs_to: other

  validate :something_involving_other

  def something_involving_other
    other.blah_blah_blah
  end
end

class Other < ApplicationRecord
end
```

When I bulk insert a parent with many children, it goes to the database for each child re-selecting "other", causing an N+1 issue, even if that association is cached when I pass it to `import!`.

```ruby
other = Other.first
parents = 5.times.map do
  Parent.new(children: 5.times.map { Child.new(other: other) })
end
parents.first.children.first.other # <- does not cause a db query
Parent.import!(parents, recursive: true) # <- causes 25 db queries for "other"
```

That is simplified code, of course.

Should this be happening? Can I avoid it?

**Comment by eric-simonton-sama** [comment]
We currently have `activerecord (5.1.4)`, and `activerecord-import (0.21.0)`

**Comment by jkowens** [comment]
This has been fixed in release v0.22.0.

**Comment by eric-simonton-sama** [comment]
Thank you!
